src/NorthwindWebsite/Middleware/ImageCachingMiddleware.cs

```diff
@@ -0,0 +1,97 @@
+﻿using NorthwindWebsite.Business.Services.Interfaces;
+using NorthwindWebsite.Core.ApplicationSettings;
+using NorthwindWebsite.Core.Constants;
+using NorthwindWebsite.Core.Utils;
+
+namespace NorthwindWebsite.Middleware;
+
+public class ImageCachingMiddleware : IMiddleware
+{
+    private const string ImagesUrlSubstring = "/images/";
+
+    private readonly AppSettings _appSettings;
+    private readonly IImageCachingService _imageCachingService;
+    private readonly ILogger<ImageCachingMiddleware> _logger;
+
+    public ImageCachingMiddleware(
+        AppSettings appSettings,
+        IImageCachingService cacheService,
+        ILogger<ImageCachingMiddleware> logger)
+    {
+        _appSettings = appSettings;
+        _imageCachingService = cacheService;
+        _logger = logger;
+    }
+
+    public async Task InvokeAsync(HttpContext context, RequestDelegate next)
+    {
+        DateTime.TryParse(context.Response.Headers["expires"], out var expires);
+
+        var requestPath = context.Request.Path.Value;
+
+        string imageIndex = requestPath!.Contains(ImagesUrlSubstring)
+            ? requestPath!.GetImageIndexFromRequest()
+            : string.Empty;
+
+        if (requestPath!.Contains(ImagesUrlSubstring)
+            && _imageCachingService.IsContained(imageIndex))
+        {
+            var imageAsBytes = _imageCachingService.GetImageFromCache(Convert.ToInt32(imageIndex));
```

Comment:
Well, this method is too huge, so, let's split it into sub-methods per each condition